services/digest_io.py

```
@@ -0,0 +1,249 @@
+                digest = json.load(f)
+            
+            # Apply AI enhancements
+            from .ai_inserts import AIInsertsService
+            ai_service = AIInsertsService()
+            
+            # Enhance digest with AI content
+            enhanced_digest = self._enhance_digest_with_ai(digest, ai_service)
+            
+            # Save as FINAL version using atomic write
+            final_path = self.blogs_dir / target_date / f"FINAL-{target_date}_digest.json"
+            cache_manager.atomic_write_json(final_path, enhanced_digest, overwrite=True)
```

Ruff (0.12.2)

Undefined name `cache_manager`

(F821)